src/clr-angular/timeline/_timeline.clarity.scss

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+// This software is released under MIT license.
+// The full license information can be found in LICENSE in the root directory of this project.
+
+@include exports('timeline.clarity') {
+  .clr-timeline {
+    display: flex;
+    padding: $clr-timeline-padding-width;
+  }
+
+  .clr-timeline-step {
+    display: flex;
+    position: relative;
```

Comment:
This might be a problem, it will cause clipping on anything like a dropdown menu. Have you tried any other options that would get around this? @hippee-lee any additional insights based on popover work?

Reply:
I did some experiments and the horizontal timeline still works fine without step (relative) and step-body:before (absolute).

But if I remove those, the vertical timeline doesn't work. The "line" is not displayed at all between steps. Or with only step-body::before (absolution) it won't be able to get its parent's (timeline-step) height by using 100%, instead it get's the full height of the entire timeline..Not sure what should be the solution to get rid of it...